### ISSUE TITLE
Cannot read property 'beta.screwdriver.cd/cpu' of undefined

### ISSUE THREAD

**Issue by stjohnjohnson** [screwdriver-cd/screwdriver#1094]
**What happened**:
Started a build in a repository (without a config) and got that error on the screen.  Build did NOT start.

**What you expected to happen**:
No strange error like that.

**How to reproduce it**:
Not sure, it happens everytime for me.

**Comment by stjohnjohnson** [comment]
More details:
```
[sd-api-5b5d5cc74f-92vw2] 180531/211016.842, (1527801012964:sd-api-5b5d5cc74f-92vw2:27:jhv13r8u:10314) [request,server,error] data: TypeError: Cannot read property 'beta.screwdriver.cd/cpu' of undefined
[sd-api-5b5d5cc74f-92vw2] at K8sExecutor._start (/usr/src/app/node_modules/screwdriver-executor-k8s/index.js:175:38)
[sd-api-5b5d5cc74f-92vw2] at validate.then.validConfig (/usr/src/app/node_modules/screwdriver-executor-base/index.js:49:39)
[sd-api-5b5d5cc74f-92vw2] at <anonymous>
```

**Comment by Filbird** [comment]
Hrmm, I can't seem to recreate that error. 

I have a pipeline created from an empty repository. The build is able to run and successfully gets to the `config-parse-error` step.
https://cd.screwdriver.cd/pipelines/934/builds/39931

Are you using `executor-queue`? Also what version of the API are you using?

**Comment by stjohnjohnson** [comment]
I'm using executor-k8s, not queue.

```json
"versions": [
"screwdriver-api@0.5.345",
"screwdriver-artifact-bookend@1.1.16",
"screwdriver-build-bookend@2.3.0",
"screwdriver-command-validator@1.0.4",
"screwdriver-config-parser@4.1.0",
"screwdriver-coverage-base@1.0.5",
"screwdriver-coverage-bookend@1.0.0",
"screwdriver-coverage-sonar@1.0.5",
"screwdriver-data-schema@18.20.0",
"screwdriver-datastore-base@3.0.6",
"screwdriver-datastore-sequelize@4.1.1",
"screwdriver-executor-base@6.1.0",
"screwdriver-executor-docker@2.3.4",
"screwdriver-executor-k8s-vm@2.4.1",
"screwdriver-executor-k8s@11.1.1",
"screwdriver-executor-nomad@1.0.4",
"screwdriver-executor-queue@2.1.9",
"screwdriver-executor-router@1.0.5",
"screwdriver-models@27.2.3",
"screwdriver-node-circuitbreaker@1.1.1",
"screwdriver-notifications-base@3.0.0",
"screwdriver-notifications-email@1.1.4",
"screwdriver-notifications-slack@2.1.2",
"screwdriver-scm-base@4.1.0",
"screwdriver-scm-github@7.1.0",
"screwdriver-scm-router@3.0.0",
"screwdriver-template-validator@3.0.5",
"screwdriver-workflow-parser@1.4.2"
]
```

Here is my config (removed any secrets):
```yaml
"auth": 
  "cookiePassword": "COOKIE"
  "encryptionPassword": "ENCRYPT"
  "https": true
  "jwtPrivateKey": "MY PRIVATE KEY"
  "jwtPublicKey": "MY PUBLIC KEY"
"datastore": 
  "plugin": "sequelize"
  "sequelize": 
    "database": "screwdriver"
    "dialect": "postgres"
    "host": "REMOTEHOST"
    "password": "PASSSSWORDDD"
    "port": "5432"
    "username": "screwdriver"
"ecosystem": 
  "store": "https://store.mydomain.com"
  "ui": "https://ui.mydomain.com"
"executor": 
  "k8s": 
    "enabled": true
    "options": 
      "kubernetes": 
        "jobsNamespace": "sd-builds"
        "resources": 
          "cpu": 
            "high": 4
            "low": 0.5
          "memory": 
            "high": 6
            "low": 2
      "launchVersion": "v4.0.101"
  "plugin": "k8s"
"httpd": 
  "host": "0.0.0.0"
  "port": 80
  "tls": false
  "uri": "https://api.mydomain.com"
"scms": 
  "github": 
    "config": 
      "oauthClientId": "ID"
      "oauthClientSecret": "SECRET"
      "privateRepo": true
      "secret": "MORESECRET"
    "plugin": "github"
```

**Comment by Filbird** [comment]
Looks like the reason `annotations` is `undefined` is because we do not default to an empty object here.
 
https://github.com/screwdriver-cd/models/blob/master/lib/build.js#L223

**Comment by jithin1987** [comment]
Thanks @Filbird for the quick fix.

**Comment by stjohnjohnson** [comment]
@jithin1987 @Filbird what version of the API is it fixed in?

**Comment by Filbird** [comment]
👍 

Should be available starting `v0.5.356`
